### ISSUE TITLE
Typecheck errors when calling a function in another go file in the same package.

### ISSUE THREAD

**Issue by aaronmcohen** [fatih/vim-go#3623]
### What did you do? (required: The issue will be **closed** when not provided)

steps:
- Create a go module project
  - cd ${GOPATH}/src
  - mkdir test
  - cd test
  - go mod init test
- create a function in the `main` package outside the `main.go` file in the `outside.go` file
```go
package main

import "fmt"

// Outside function demonstrates a simple function outside the main.go file
func Outside() {
	fmt.Println("outside")
}
```
- create a `main.go` which calls the function in `outside.go`
```go
package main

func main() {
	Outside()
}
```

### What did you expect to happen?

I expected the no errors in my Quickfix list

### What happened instead?

Any call to a function in another file in the same package will fail with a `typecheck` error.

```go
package main
 
 func main() {
   Outside() // E: undefined: Outside (typecheck)
 }
       
```

> 1 main.go|4 col 2 error| undefined: Outside (typecheck)

####  `vimrc` you used to reproduce:

<details><summary>vimrc</summary>

```vim
"""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""
" => General
"""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""
set number

" Sets how many lines of history VIM has to remember
set history=500

" Enable filetype plugins
filetype plugin on
filetype indent on

" Set to auto read when a file is changed from the outside
set autoread
au FocusGained,BufEnter * checktime

let $LANG='en'
set langmenu=en
source $VIMRUNTIME/delmenu.vim
source $VIMRUNTIME/menu.vim

" Turn on the Wild menu
set wildmenu

" Ignore compiled files
set wildignore=*.o,*~,*.pyc
if has("win16") || has("win32")
    set wildignore+=.git\*,.hg\*,.svn\*
else
    set wildignore+=*/.git/*,*/.hg/*,*/.svn/*,*/.DS_Store
endif

"Always show current position
set ruler

" Height of the command bar
set cmdheight=1

" A buffer becomes hidden when it is abandoned
set hid

" Ignore case when searching
set ignorecase

" When searching try to be smart about cases 
set smartcase

" Highlight search results
set hlsearch

" Makes search act like search in modern browsers
set incsearch

" Don't redraw while executing macros (good performance config)
set lazyredraw

" For regular expressions turn magic on
set magic

" Show matching brackets when text indicator is over them
set showmatch
" How many tenths of a second to blink when matching brackets
set mat=2

" Add a bit extra margin to the left
set foldcolumn=1

" Set extra options when running in GUI mode
if has("gui_running")
    set guioptions-=T
    set guioptions-=e
    set t_Co=256
    set guitablabel=%M\ %t
endif

" Set utf8 as standard encoding and en_US as the standard language
set encoding=utf8

" Use Unix as the standard file type
set ffs=unix,dos,mac

"""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""
" => Files, backups and undo
"""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""
" Turn backup off, since most stuff is in SVN, git etc. anyway...
set nobackup
set nowb
set noswapfile

"""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""
" => Text, tab and indent related
"""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""
" Use spaces instead of tabs
set expandtab

" Be smart when using tabs ;)
set smarttab

" 1 tab == 2 spaces
set shiftwidth=2
set tabstop=2
set softtabstop=2

" Linebreak on 500 characters
set lbr
set tw=500

set ai "Auto indent
set si "Smart indent
set wrap "Wrap lines

" Return to last edit position when opening files (You want this!)
au BufReadPost * if line("'\"") > 1 && line("'\"") <= line("$") | exe "normal! g'\"" | endif

""""""""""""""""""""""""""""""
" => Status line
""""""""""""""""""""""""""""""
" Always show the status line
set laststatus=2

" Format the status line
set statusline=\ %{HasPaste()}%F%m%r%h\ %w\ \ CWD:\ %r%{getcwd()}%h\ \ \ Line:\ %l\ \ Column:\ %c

" Delete trailing white space on save, useful for some filetypes ;)
fun! CleanExtraSpaces()
    let save_cursor = getpos(".")
    let old_query = getreg('/')
    silent! %s/\s\+$//e
    call setpos('.', save_cursor)
    call setreg('/', old_query)
endfun

if has("autocmd")
    autocmd BufWritePre *.txt,*.js,*.py,*.wiki,*.sh,*.coffee,*.yaml,*.yml :call CleanExtraSpaces()
endif

"""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""
" => Spell checking
"""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""
" Pressing ,ss will toggle and untoggle spell checking
map <leader>ss :setlocal spell!<cr>

"""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""
" => Helper functions
"""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""
" Returns true if paste mode is enabled
function! HasPaste()
    if &paste
        return 'PASTE MODE  '
    endif
    return ''
endfunction

" Don't close window, when deleting a buffer
command! Bclose call <SID>BufcloseCloseIt()
function! <SID>BufcloseCloseIt()
    let l:currentBufNum = bufnr("%")
    let l:alternateBufNum = bufnr("#")

    if buflisted(l:alternateBufNum)
        buffer #
    else
        bnext
    endif

    if bufnr("%") == l:currentBufNum
        new
    endif

    if buflisted(l:currentBufNum)
        execute("bdelete! ".l:currentBufNum)
    endif
endfunction

function! CmdLine(str)
    call feedkeys(":" . a:str)
endfunction 

function! VisualSelection(direction, extra_filter) range
    let l:saved_reg = @"
    execute "normal! vgvy"

    let l:pattern = escape(@", "\\/.*'$^~[]")
    let l:pattern = substitute(l:pattern, "\n$", "", "")

    if a:direction == 'gv'
        call CmdLine("Ack '" . l:pattern . "' " )
    elseif a:direction == 'replace'
        call CmdLine("%s" . '/'. l:pattern . '/')
    endif

    let @/ = l:pattern
    let @" = l:saved_reg
endfunction

"""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""
" => GUI related
"""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""
" Set font according to system
set gfn=IBM\ Plex\ Mono:h14,Hack:h14,Source\ Code\ Pro:h15,Menlo:h15

"---
syntax enable
set path+=**

" Fix Delete key behavior
set backspace=indent,eol,start
set whichwrap+=<,>,h,l

" Mouse Support
set nocompatible
set mouse=a

" FixCopy Paste
set clipboard=unnamed
" ---- Windows Like keys ----
" CTRL-Z is Undo; not in cmdline though
noremap <C-Z> u
inoremap <C-Z> <C-O>u
" CTRL-Y is Redo (although not repeat); not in cmdline though
noremap <C-Y> <C-R>
inoremap <C-Y> <C-O><C-R>
" CTRL-A is Select all
noremap <C-A> gggH<C-O>G
inoremap <C-A> <C-O>gg<C-O>gH<C-O>G
cnoremap <C-A> <C-C>gggH<C-O>G
" CTRL-F4 is Close window
noremap <C-F4> <C-W>c
inoremap <C-F4> <C-O><C-W>c
cnoremap <C-F4> <C-C><C-W>c
" CTRL-Tab is Next window
noremap <C-Tab> <C-W>w
inoremap <C-Tab> <C-O><C-W>w
cnoremap <C-Tab> <C-C><C-W>w
" ---- Windows Like Copy-Paste keys ----
" CTRL-v is paste
inoremap <C-v> <esc>"*p<return>i
noremap <C-v> "*p<return>
" CTRL-x is cut (in visual mode only)
vnoremap <C-x> "*d
" CTRL-c is copy (in visual mode only)
vnoremap <C-c> "*y
" ---- Restore some remapped things 
" make real <C-V> (visual block) as <C-Q> available
noremap <c-q> <c-v>
inoremap <C-Y> <C-Y>

" Specify a directory for plugins
" - For Neovim: stdpath('data') . '/plugged'
" - Avoid using standard Vim directory names like 'plugin'
call plug#begin('~/.vim/plugged')

" Make sure you use single quotes

" Yaml Formatter
Plug 'tarekbecker/vim-yaml-formatter'
" Syntax checking
Plug 'scrooloose/syntastic'
" Rust Syntax support
Plug 'rust-lang/rust.vim'
Plug 'rust-lang/rls'
Plug 'rhysd/rust-doc.vim'
" File navigation
Plug 'scrooloose/nerdtree'
" Git diff in the sign column
Plug 'airblade/vim-gitgutter'
" Git support
Plug 'tpope/vim-fugitive'
" Browse CTAGS of the current file
Plug 'majutsushi/tagbar'
" tender color scheme
"Plug 'jacoborus/tender.vim'
" status/tabline 
Plug 'vim-airline/vim-airline'
" VIM Markdown support
Plug 'godlygeek/tabular'
Plug 'plasticboy/vim-markdown'
" Linting
Plug 'dense-analysis/ale'

" VIM Autocomplete
Plug 'prabirshrestha/async.vim'
Plug 'prabirshrestha/vim-lsp'
Plug 'mattn/vim-lsp-settings'
Plug 'prabirshrestha/asyncomplete.vim'
Plug 'prabirshrestha/asyncomplete-lsp.vim'
Plug 'andreypopp/asyncomplete-ale.vim'
Plug 'htlsne/asyncomplete-look'
Plug 'prabirshrestha/asyncomplete-file.vim'
Plug 'prabirshrestha/asyncomplete-gocode.vim'
Plug 'Shougo/unite.vim'

" PlantUML
Plug 'aklt/plantuml-syntax'
Plug 'weirongxu/plantuml-previewer.vim'
Plug 'tyru/open-browser.vim'

"Nord THeme https://www.nordtheme.com/docs/ports/vim/installation
Plug 'arcticicestudio/nord-vim'

" Markdown preview
Plug 'iamcco/markdown-preview.nvim', { 'do': 'cd app && yarn install'  }
" https://github.com/junegunn/vim-easy-align
Plug 'junegunn/vim-easy-align'

" Add maktaba and bazel to the runtimepath.
" (The latter must be installed before it can be used.)
Plug 'google/vim-maktaba'
Plug 'bazelbuild/vim-bazel'

" Go support https://github.com/fatih/vim-go
Plug 'fatih/vim-go', { 'do': ':GoUpdateBinaries' }

Plug 'skanehira/k8s.vim'
Plug 'vim-denops/denops.vim'

Plug 'aliou/bats.vim'

Plug 'pearofducks/ansible-vim'

Plug 'mattn/vim-goimpl'

" Initialize plugin system
call plug#end()


" Align GitHub-flavored Markdown tables
au FileType markdown vmap <Leader><Bslash> :EasyAlign*<Bar><Enter>

" Treat bats file like bash files
" Disable sh shellchecker to avoid false errors
" shellcheck will catch all errors
au BufRead,BufNewFile *.bats set ft=bash |
  \ let g:syntastic_sh_checkers = ['shellcheck']

" Recommended Syntastic settings
set statusline+=%#warningmsg#
set statusline+=%{SyntasticStatuslineFlag()}
set statusline+=%*
let g:syntastic_always_populate_loc_list = 1
let g:syntastic_auto_loc_list = 1
let g:syntastic_check_on_open = 1
let g:syntastic_check_on_wq = 0

" Nerdree configuration
" autocmd vimenter * NERDTree
map <C-n> :NERDTreeToggle<CR>
let NERDTreeShowHidden=1

" GitGutter configuration
autocmd BufWritePost * GitGutter


" Autocomplete configuration
let g:asyncomplete_auto_popup = 1
let g:asyncomplete_auto_completeopt = 0
set completeopt=menuone,noinsert,noselect,preview
autocmd! CompleteDone * if pumvisible() == 0 | pclose | endif
imap <c-space> <Plug>(asyncomplete_force_refresh)
if executable('rls')
    au User lsp_setup call lsp#register_server({
        \ 'name': 'rls',
        \ 'cmd': {server_info->['rustup', 'run', 'stable', 'rls']},
        \ 'whitelist': ['rust'],
        \ })
endif
if executable('pyls')
    " pip install python-language-server
    au User lsp_setup call lsp#register_server({
        \ 'name': 'pyls',
        \ 'cmd': {server_info->['pyls']},
        \ 'allowlist': ['python'],
        \ })
endif
au User asyncomplete_setup call asyncomplete#ale#register_source({
    \ 'name': 'reason',
    \ 'linter': 'flow',
    \ })
au User asyncomplete_setup call asyncomplete#register_source({
    \ 'name': 'look',
    \ 'allowlist': ['text', 'markdown'],
    \ 'completor': function('asyncomplete#sources#look#completor'),
    \ })
au User asyncomplete_setup call asyncomplete#register_source(asyncomplete#sources#file#get_source_options({
    \ 'name': 'file',
    \ 'allowlist': ['*'],
    \ 'priority': 10,
    \ 'completor': function('asyncomplete#sources#file#completor')
    \ }))
call asyncomplete#register_source(asyncomplete#sources#gocode#get_source_options({
    \ 'name': 'gocode',
    \ 'allowlist': ['go'],
    \ 'completor': function('asyncomplete#sources#gocode#completor'),
    \ 'config': {
    \    'gocode_path': expand('~/go/bin/gocode')
    \  },
    \ }))
" Linting configuraiton
let g:ale_sign_column_always = 1
let g:airline#extensions#ale#enabled = 1
let g:ale_set_quickfix = 1
let g:ale_open_list = 1

" Tender color scheme settings
"colorscheme tender

" Nord Color Scheme https://www.nordtheme.com/docs/ports/vim/installation
colorscheme nord

"let g:airline_theme = 'tender'
let g:airline_theme = 'nord'
highlight link SyntasticError SpellBad
highlight link SyntasticWarning SpellCap

" https://github.com/tarekbecker/vim-yaml-formatter
let g:yaml_formatter_indent_collection=1
autocmd FileType yaml setlocal ts=2 sts=2 sw=2 expandtab

"Vim quickfix from https://github.com/fatih/vim-go-tutorial#fix-it
set autowrite
let g:go_metalinter_command='golangci-lint'
let g:go_debug=['shell-commands','lsp']
let g:go_debug_windows = {
          \ 'vars':       'leftabove 30vnew',
          \ 'stack':      'leftabove 20new',
          \ 'goroutines': 'botright 10new',
          \ 'out':        'botright 5new',
\ }
let g:go_def_mode='gopls'
let g:go_info_mode='gopls'
let g:go_implements_mode='gopls'
let g:go_referrers_mode = 'gopls'
```
</details>

#### Vim version (first three lines from `:version`):

VIM - Vi IMproved 9.0 (2022 Jun 28, compiled Dec 05 2023 14:38:33)
macOS version - x86_64
Included patches: 1-2150

#### Go version (`go version`):

go version go1.21.5 darwin/amd64

#### Go environment
<details><summary><code>go env</code> Output:</summary><br><pre>

GO111MODULE='auto'
GOARCH='amd64'
GOBIN=''
GOCACHE='/Users/amcohen@us.ibm.com/Library/Caches/go-build'
GOENV='/Users/amcohen@us.ibm.com/Library/Application Support/go/env'
GOEXE=''
GOEXPERIMENT=''
GOFLAGS=''
GOHOSTARCH='amd64'
GOHOSTOS='darwin'
GOINSECURE=''
GOMODCACHE='/Users/amcohen@us.ibm.com/Projects/go/pkg/mod'
GONOPROXY='github.ibm.com'
GONOSUMDB='github.ibm.com'
GOOS='darwin'
GOPATH='/Users/amcohen@us.ibm.com/Projects/go'
GOPRIVATE='github.ibm.com'
GOPROXY='https://proxy.golang.org,direct'
GOROOT='/usr/local/Cellar/go/1.21.5/libexec'
GOSUMDB='sum.golang.org'
GOTMPDIR=''
GOTOOLCHAIN='auto'
GOTOOLDIR='/usr/local/Cellar/go/1.21.5/libexec/pkg/tool/darwin_amd64'
GOVCS=''
GOVERSION='go1.21.5'
GCCGO='gccgo'
GOAMD64='v1'
AR='ar'
CC='cc'
CXX='c++'
CGO_ENABLED='1'
GOMOD='/Users/amcohen@us.ibm.com/Projects/go/pkg/test/go.mod'
GOWORK=''
CGO_CFLAGS='-O2 -g'
CGO_CPPFLAGS=''
CGO_CXXFLAGS='-O2 -g'
CGO_FFLAGS='-O2 -g'
CGO_LDFLAGS='-O2 -g'
PKG_CONFIG='pkg-config'
GOGCCFLAGS='-fPIC -arch x86_64 -m64 -pthread -fno-caret-diagnostics -Qunused-arguments -fmessage-length=0 -ffile-prefix-map=/var/folders/6y/t5vfqykd33l1xxq5f9kvrx3m0000gn/T/go-build3382877517=/tmp/go-build -gno-record-gcc-switches -fno-common'

</pre></details>

#### gopls version
<details><summary><code>gopls version</code> Output:</summary><br><pre>

golang.org/x/tools/gopls v0.14.2
    golang.org/x/tools/gopls@v0.14.2 h1:sIw6vjZiuQ9S7s0auUUkHlWgsCkKZFWDHmrge8LYsnc=

</pre></details>

#### vim-go configuration:
<details><summary>vim-go configuration</summary><br><pre>
g:go_debug_windows = {'vars': 'leftabove 30vnew', 'out': 'botright 5new', 'stack': 'leftabove 20new', 'goroutines': 'botright 10new'}
g:go_metalinter_command = 'golangci-lint'
g:go_loaded_gosnippets = 1
g:go_debug = ['shell-commands', 'lsp']
g:go_def_mode = 'gopls'
g:go_info_mode = 'gopls'
g:go_gotags_bin = 'gotags'
g:go_jump_to_error = 1
g:go_implements_mode = 'gopls'
g:go_referrers_mode = 'gopls'
g:go_loaded_install = 1
</pre></details>

#### filetype detection configuration:
<details><summary>filetype detection</summary><br><pre>
filetype detection:ON  plugin:ON  indent:ON
</pre></details>

#### Additional Debug information

lsp:
<details><pre>
===== sent =====
Content-Length: 741

{"method":"initialize","jsonrpc":"2.0","id":1,"params":{"rootUri":"file:///Users/amcohen%40us.ibm.com/Projects/go/pkg/test","capabilities":{"workspace":{"workspaceFolders":true,"workspaceEdit":{"documentChanges":true},"configuration":true,"didChangeConfiguration":{"dynamicRegistration":true}},"textDocument":{"codeAction":{"codeActionLiteralSupport":{"codeActionKind":{"valueSet":["source.organizeImports","refactor.rewrite","refactor.extract"]}}},"completion":{"completionItem":{"snippetSupport":false}},"hover":{"contentFormat":["plaintext"]}}},"clientInfo":{"name":"vim-go"},"processId":21710,"workspaceFolders":[{"uri":"file:///Users/amcohen%40us.ibm.com/Projects/go/pkg/test","name":"/Users/amcohen@us.ibm.com/Projects/go/pkg/test"}]}}
===== stderr =====
serve.go:443: debug server listening at http://localhost:57506
===== received =====
Content-Length: 280

{"jsonrpc":"2.0","method":"window/logMessage","params":{"type":3,"message":"2023/12/27 21:09:40 Handshake session update\n\tupdate_session=1\n\tdebug_address=\"127.0.0.1:57506\"\n\tlogfile=\"\"\n\tserver=\"1\"\n\tgopls_path=\"/Users/amcohen@us.ibm.com/Projects/go/bin/gopls\"\n"}}
===== received =====
Content-Length: 4538

{"jsonrpc":"2.0","result":{"capabilities":{"textDocumentSync":{"change":2,"openClose":true,"save":{}},"completionProvider":{"triggerCharacters":["."]},"hoverProvider":true,"signatureHelpProvider":{"triggerCharacters":["(",","]},"definitionProvider":true,"typeDefinitionProvider":true,"implementationProvider":true,"referencesProvider":true,"documentHighlightProvider":true,"documentSymbolProvider":true,"codeActionProvider":{"codeActionKinds":["quickfix","refactor.extract","refactor.inline","refactor.rewrite","source.fixAll","source.organizeImports"]},"codeLensProvider":{},"documentLinkProvider":{},"workspaceSymbolProvider":true,"documentFormattingProvider":true,"renameProvider":true,"foldingRangeProvider":true,"selectionRangeProvider":true,"executeCommandProvider":{"commands":["gopls.add_dependency","gopls.add_import","gopls.add_telemetry_counters","gopls.apply_fix","gopls.change_signature","gopls.check_upgrades","gopls.edit_go_directive","gopls.fetch_vulncheck_result","gopls.gc_details","gopls.generate","gopls.go_get_package","gopls.list_imports","gopls.list_known_packages","gopls.maybe_prompt_for_telemetry","gopls.mem_stats","gopls.regenerate_cgo","gopls.remove_dependency","gopls.reset_go_mod_diagnostics","gopls.run_go_work_command","gopls.run_govulncheck","gopls.run_tests","gopls.start_debugging","gopls.start_profile","gopls.stop_profile","gopls.test","gopls.tidy","gopls.toggle_gc_details","gopls.update_go_sum","gopls.upgrade_dependency","gopls.vendor","gopls.workspace_stats"]},"callHierarchyProvider":true,"semanticTokensProvider":{"full":true,"legend":{"tokenModifiers":[],"tokenTypes":[]},"range":true},"inlayHintProvider":{},"workspace":{"workspaceFolders":{"supported":true,"changeNotifications":"workspace/didChangeWorkspaceFolders"}}},"serverInfo":{"name":"gopls","version":"{\"GoVersion\":\"go1.21.5\",\"Path\":\"golang.org/x/tools/gopls\",\"Main\":{\"Path\":\"golang.org/x/tools/gopls\",\"Version\":\"v0.14.2\",\"Sum\":\"h1:sIw6vjZiuQ9S7s0auUUkHlWgsCkKZFWDHmrge8LYsnc=\",\"Replace\":null},\"Deps\":[{\"Path\":\"github.com/BurntSushi/toml\",\"Version\":\"v1.2.1\",\"Sum\":\"h1:9F2/+DoOYIOksmaJFPw1tGFy1eDnIJXg+UHjuD8lTak=\",\"Replace\":null},{\"Path\":\"github.com/google/go-cmp\",\"Version\":\"v0.5.9\",\"Sum\":\"h1:O2Tfq5qg4qc4AmwVlvv0oLiVAGB7enBSJ2x2DqQFi38=\",\"Replace\":null},{\"Path\":\"github.com/sergi/go-diff\",\"Version\":\"v1.1.0\",\"Sum\":\"h1:we8PVUC3FE2uYfodKH/nBHMSetSfHDR6scGdBi+erh0=\",\"Replace\":null},{\"Path\":\"golang.org/x/exp/typeparams\",\"Version\":\"v0.0.0-20221212164502-fae10dda9338\",\"Sum\":\"h1:2O2DON6y3XMJiQRAS1UWU+54aec2uopH3x7MAiqGW6Y=\",\"Replace\":null},{\"Path\":\"golang.org/x/mod\",\"Version\":\"v0.14.0\",\"Sum\":\"h1:dGoOF9QVLYng8IHTm7BAyWqCqSheQ5pYWGhzW00YJr0=\",\"Replace\":null},{\"Path\":\"golang.org/x/sync\",\"Version\":\"v0.4.0\",\"Sum\":\"h1:zxkM55ReGkDlKSM+Fu41A+zmbZuaPVbGMzvvdUPznYQ=\",\"Replace\":null},{\"Path\":\"golang.org/x/sys\",\"Version\":\"v0.14.0\",\"Sum\":\"h1:Vz7Qs629MkJkGyHxUlRHizWJRG2j8fbQKjELVSNhy7Q=\",\"Replace\":null},{\"Path\":\"golang.org/x/telemetry\",\"Version\":\"v0.0.0-20231114163143-69313e640400\",\"Sum\":\"h1:brbkEFfGwNGAEkykUOcryE/JiHUMMJouzE0fWWmz/QU=\",\"Replace\":null},{\"Path\":\"golang.org/x/text\",\"Version\":\"v0.13.0\",\"Sum\":\"h1:ablQoSUd0tRdKxZewP80B+BaqeKJuVhuRxj/dkrun3k=\",\"Replace\":null},{\"Path\":\"golang.org/x/tools\",\"Version\":\"v0.14.1-0.20231114185516-c9d3e7de13fd\",\"Sum\":\"h1:Oku7E+OCrXHyst1dG1z10etCTxewCHXNFLRlyMPbh3w=\",\"Replace\":null},{\"Path\":\"golang.org/x/vuln\",\"Version\":\"v1.0.1\",\"Sum\":\"h1:KUas02EjQK5LTuIx1OylBQdKKZ9jeugs+HiqO5HormU=\",\"Replace\":null},{\"Path\":\"honnef.co/go/tools\",\"Version\":\"v0.4.5\",\"Sum\":\"h1:YGD4H+SuIOOqsyoLOpZDWcieM28W47/zRO7f+9V3nvo=\",\"Replace\":null},{\"Path\":\"mvdan.cc/gofumpt\",\"Version\":\"v0.4.0\",\"Sum\":\"h1:JVf4NN1mIpHogBj7ABpgOyZc65/UUOkKQFkoURsz4MM=\",\"Replace\":null},{\"Path\":\"mvdan.cc/xurls/v2\",\"Version\":\"v2.4.0\",\"Sum\":\"h1:tzxjVAj+wSBmDcF6zBB7/myTy3gX9xvi8Tyr28AuQgc=\",\"Replace\":null}],\"Settings\":[{\"Key\":\"-buildmode\",\"Value\":\"exe\"},{\"Key\":\"-compiler\",\"Value\":\"gc\"},{\"Key\":\"DefaultGODEBUG\",\"Value\":\"panicnil=1\"},{\"Key\":\"CGO_ENABLED\",\"Value\":\"1\"},{\"Key\":\"CGO_CFLAGS\",\"Value\":\"\"},{\"Key\":\"CGO_CPPFLAGS\",\"Value\":\"\"},{\"Key\":\"CGO_CXXFLAGS\",\"Value\":\"\"},{\"Key\":\"CGO_LDFLAGS\",\"Value\":\"\"},{\"Key\":\"GOARCH\",\"Value\":\"amd64\"},{\"Key\":\"GOOS\",\"Value\":\"darwin\"},{\"Key\":\"GOAMD64\",\"Value\":\"v1\"}],\"Version\":\"v0.14.2\"}"}},"id":1}
===== sent =====
Content-Length: 52

{"method":"initialized","jsonrpc":"2.0","params":{}}
===== sent =====
Content-Length: 230

{"method":"workspace/didChangeWorkspaceFolders","jsonrpc":"2.0","params":{"event":{"added":[{"uri":"file:///Users/amcohen%40us.ibm.com/Projects/go/pkg/test","name":"/Users/amcohen@us.ibm.com/Projects/go/pkg/test"}],"removed":[]}}}
===== sent =====
Content-Length: 236

{"method":"textDocument/didOpen","jsonrpc":"2.0","params":{"textDocument":{"uri":"file:///Users/amcohen%40us.ibm.com/Projects/go/pkg/test/main.go","version":2,"languageId":"go","text":"package main\n\nfunc main() {\n\tOutside()\n}\n"}}}
===== received =====
Content-Length: 99

{"jsonrpc":"2.0","method":"window/showMessage","params":{"type":4,"message":"Loading packages..."}}
===== received =====
Content-Length: 167

{"jsonrpc":"2.0","method":"workspace/configuration","params":{"items":[{"scopeUri":"file:///Users/amcohen@us.ibm.com/Projects/go/pkg/test","section":"gopls"}]},"id":1}
===== sent =====
Content-Length: 78

{"id":1,"jsonrpc":"2.0","result":[{"buildFlags":[],"hoverKind":"Structured"}]}
===== received =====
Content-Length: 611

{"jsonrpc":"2.0","method":"window/logMessage","params":{"type":3,"message":"2023/12/27 21:09:40 go info for /Users/amcohen@us.ibm.com/Projects/go/pkg/test\n(go dir /Users/amcohen@us.ibm.com/Projects/go/pkg/test)\n(go version go version go1.21.5 darwin/amd64)\n(valid build configuration = true)\n(build flags: [])\n(selected go env: [GO111MODULE=auto, GOCACHE=/Users/amcohen@us.ibm.com/Library/Caches/go-build, GOFLAGS=, GOMODCACHE=/Users/amcohen@us.ibm.com/Projects/go/pkg/mod, GOPATH=/Users/amcohen@us.ibm.com/Projects/go, GOPRIVATE=github.ibm.com, GOROOT=/usr/local/Cellar/go/1.21.5/libexec, GOWORK=])\n\n"}}
===== received =====
Content-Length: 285

{"jsonrpc":"2.0","method":"window/logMessage","params":{"type":3,"message":"2023/12/27 21:09:40 go/packages.Load #1\n\tsnapshot=0\n\tdirectory=file:///Users/amcohen@us.ibm.com/Projects/go/pkg/test\n\tquery=[/Users/amcohen@us.ibm.com/Projects/go/pkg/test/... builtin]\n\tpackages=2\n"}}
===== received =====
Content-Length: 155

{"jsonrpc":"2.0","method":"window/logMessage","params":{"type":3,"message":"2023/12/27 21:09:40 go/packages.Load #1: updating metadata for 42 packages\n"}}
===== received =====
Content-Length: 106

{"jsonrpc":"2.0","method":"window/showMessage","params":{"type":3,"message":"Finished loading packages."}}
===== received =====
Content-Length: 176

{"jsonrpc":"2.0","method":"client/registerCapability","params":{"registrations":[{"id":"workspace/didChangeConfiguration","method":"workspace/didChangeConfiguration"}]},"id":2}
===== sent =====
</pre></details>

messages:
```
Messages maintainer: The Vim Project
"main.go" 5L, 41B
vim-go: shell command: 'go' 'env' 'GOBIN'
vim-go: shell command: '/usr/local/bin/go' 'list'
vim-go: job command: ['/Users/amcohen@us.ibm.com/Projects/go/bin/gopls', '-debug', 'localhost:0', '-remote=auto']
vim-go: shell command: '/usr/local/bin/go' 'env' 'GOMOD'
vim-go: shell command: '/usr/local/bin/go' 'env' 'GOMOD'
vim-go: initializing gopls
vim-go: initialized gopls
vim-go: Finished loading packages.
```

gopls -rpc.trace -v check main.go           
```
2023/12/27 21:22:53 Info:2023/12/27 21:22:53 go info for /Users/amcohen@us.ibm.com/Projects/go/pkg/test
(go dir /Users/amcohen@us.ibm.com/Projects/go/pkg/test)
(go version go version go1.21.5 darwin/amd64)
(valid build configuration = true)
(build flags: [])
(selected go env: [GO111MODULE=auto, GOCACHE=/Users/amcohen@us.ibm.com/Library/Caches/go-build, GOFLAGS=, GOMODCACHE=/Users/amcohen@us.ibm.com/Projects/go/pkg/mod, GOPATH=/Users/amcohen@us.ibm.com/Projects/go, GOPRIVATE=github.ibm.com, GOROOT=/usr/local/Cellar/go/1.21.5/libexec, GOWORK=])

2023/12/27 21:22:53 Info:2023/12/27 21:22:53 go/packages.Load #1
	snapshot=0
	directory=file:///Users/amcohen@us.ibm.com/Projects/go/pkg/test
	query=[/Users/amcohen@us.ibm.com/Projects/go/pkg/test/... builtin]
	packages=2
2023/12/27 21:22:53 Info:2023/12/27 21:22:53 go/packages.Load #1: updating metadata for 42 packages
```

go build -v -a 
```
internal/goarch
internal/godebugs
internal/goexperiment
unicode/utf8
unicode
internal/unsafeheader
sync/atomic
internal/cpu
runtime/internal/math
runtime/internal/atomic
internal/abi
internal/itoa
internal/goos
internal/race
math/bits
internal/coverage/rtcov
runtime/internal/sys
internal/bytealg
math
runtime
internal/reflectlite
sync
internal/testlog
errors
sort
internal/oserror
internal/safefilepath
path
io
strconv
syscall
reflect
internal/syscall/execenv
internal/syscall/unix
time
io/fs
internal/poll
internal/fmtsort
os
fmt
test
```

**Comment by lkramer** [comment]
I'm seeing the same thing...

**Comment by bhcleek** [comment]
Why are you creating a module in `$GOPATH/pkg`? Does the problem you're seeing happen if you delete that directory?

```
cd ${GOPATH}/pkg
mkdir test
cd test
mod init test
```

Can you clarify what this means?
> create a function in the main package outside the main.go file in the outside.go file

**Comment by aaronmcohen** [comment]
@bhcleek Yeah this originally happened in another coding project which worked great prior to upgrading vim-go. Those instructions were to just build a simple reproducible test case. Clearing out my entire ${GOPATH} does not solve the issue. Also the same steps will fail in the ${GOPATH}/src directory.

**Comment by bhcleek** [comment]
I'm not seeing this at all. What operation did you execute to cause the error?

**Comment by aaronmcohen** [comment]
oddly enough it was just `PlugUpdate` & `GoUpdateBinaries`

**Comment by aaronmcohen** [comment]
What kind of debug information can I provide which will better provide insight to the issue?

**Comment by bhcleek** [comment]
Just for clarity as an aside; Go source code should not be placed in `$GOPATH/pkg`. The `pkg` directory is intended for Go to use for a variety of its internal uses. If you're able to duplicate your issue without explicitly creating any source code files in that directory, then it's unlikely to be a cause of the problem you're seeing. But I'd advise against it as a general practice anyway.

**Comment by bhcleek** [comment]
> oddly enough it was just PlugUpdate & GoUpdateBinaries

What other plugins where updated with `PlugUpdate`?

Can you duplicate this problem when vim-go is the only plugin?

Also, vim-go isn't going to perform any operations that would cause errors to be displayed like that unless you did something. Think of it from my perspective: I don't see any vim commands or keystrokes being described here, only the creation of a file. Clearly there's some missing steps for me to be able to duplicate what you're doing.

**Comment by aaronmcohen** [comment]
Completely agree. Complete goof on my part for the project location. That being said I have reproduced it in other locations such as `${GOPATH}/src`.

So I deleted my existing `.vim` directory and did a fresh install of `vim-go` with no other plugins and it worked fine. The problem actually seems to come from the `dense-analysis/ale` plugin. I found  https://github.com/dense-analysis/ale/issues/4586 and `let g:ale_go_golangci_lint_package=1` fixed it. Thank you @bhcleek for helping and I am closing this issue.
